qa/common/gen_qa_pytorch_model.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import argparse
+import os
+
+import torch
+from torch import nn
+
+
+class AddSubNet(nn.Module):
+    def __init__(self):
+        super(AddSubNet, self).__init__()
+
+    def forward(self, input0, input1):
+        return (input0 + input1), (input0 - input1)
+
+
+def generate_model(model_dir):
+    model = AddSubNet()
+
+    traced_model = torch.jit.trace(
+        model,
+        (torch.rand(1, 4, dtype=torch.float), torch.rand(1, 4, dtype=torch.float)),
+    )
+
+    os.makedirs(model_dir, exist_ok=True)
+    model_path = os.path.join(model_dir, "model.pt")
+
+    traced_model.save(model_path)
+
+
+def generate_config(config_path):
```

Comment:
nit: I believe add_sub's config is similar to what this function is generating. Maybe we can re-use this function to generate add_sub config as well? Maybe as an input it can take backend's name and dynamically adjust produced `config` .

Reply:
@oandreeva-nv, just for confirmation, would recommend to generate [qa/python_models/python_based_backends/add_sub_backend/config.pbtxt](https://github.com/triton-inference-server/server/pull/6466/files#diff-85437ebf20c6bf63c7225f32423b65dfa4056538b204cf6d4a7e0fc10a6ba6fa) using this function?

Reply:
@pskiran1 How about we add inputs and outputs through [auto_complete_config](https://github.com/triton-inference-server/python_backend#auto_complete_config) for add_sub backend model then? In this case, we can make this function conditional, e.g.  `generate_config(config_path, backend_name, add_io = True)`. If `add_io` is false, it will only generate " backend: "{}"".format(backend_name)" something like this. Could you please try it out?

Reply:
Using auto_complete_config is a good idea here!

Reply:
Updated the add_sub backend model.py to prepare inputs/outputs configuration using the `auto-complete-config` feature and removed a separate config.pbtxt file. For simplicity, generating minimal config.pbtxt in test.sh using command,
`echo "backend: \"add_sub\"" > ${MODEL_REPOSITORY}/add/config.pbtxt`